keps/sig-cluster-lifecycle/addons/20191013-install-addons-via-kubeadm.md

```diff
@@ -0,0 +1,450 @@
+This KEP proposes an additional phase to kubeadm that invokes the addon-installer library.
+
+## Motivation
+
+<!-- This section is for explicitly listing the motivation, goals and non-goals of this KEP.
+Describe why the change is important and the benefits to users.
+The motivation section can optionally provide links to [experience reports][] to demonstrate the interest in a KEP within the wider Kubernetes community. -->
+
+Kubeadm currently installs two "core" addons into the cluster once the control-plane is running:  kube-proxy and CoreDNS.
+Optionally, it also supports installing kube-dns in place of CoreDNS.
+Installation of these addons occurs via application of manifests for which the strings are built into and versioned with the kubeadm binary.
+It is not currently possible to disable installation of these addons or modify them with a formal kubeadm init.
+
+Upgrades are accomplished with modifications to kubeadm's upgrade logic, often by introspecting which GVK's exist in the current cluster. 
+
+Kubeadm installs these "core" addons leaving an almost functioning cluster available for the user, but a CNI implementation is not installed. That is left to the cluster operator.
+
+Maintaining these manifests and this logic with kubeadm has to this point has allowed users to recieve a useful, default setup.
+As the ecosystem grows, this is proving to now be too inflexible for users and vendors.
+
+Use of ClusterAPI means less operators will be manually mutating clusters before they are expected to function.
+Kubeadm does not need to maintain CNI installation, but users should be able to use kubeadm to produce a functioning cluster, complete with with operational CNI. 
+
+### Goals
+
+<!-- List the specific goals of the KEP.
+How will we know that this has succeeded? -->
+
+Integrating the addon-installer library and API with kubeadm will allow users to opt out of installing kube-proxy or CoreDNS/kube-dns.
+Some reasons for this may be:
+- The user does not desire pod networking and/or service-discovery in their cluster.
+- CNI implementations such as kube-router and Cilium can actually replace kube-proxy.
+
+The "core" addons can be referenced and installed from a central, extensible, shared implementation.
+In the case of CoreDNS, this may even make sense to be managed by the upstream project instead of maintaining copies of manifests in kubernetes/kubernetes.
+
+This integration will also allow users to extend these "core" addons using their own kustomize driven patches.
+Patches are expected to be supplied via the API, filesystem, or git with additional potential for OCI images.
+
+This will also enable users to specify additional addons to install such as CNI implementations, drivers, webhook-servers, or other API's and operators.
```

Comment:
```suggestion
This will also enable users to specify additional addons to install such as CNI implementations, drivers, webhook-servers, or other APIs and operators.
```